src/inline_spl_token_2022.rs

```diff
@@ -0,0 +1,19 @@
+/// Partial SPL Token declarations inlined to avoid an external dependency on the spl-token-2022 crate
+/// Copied from solana-runtime
```

Comment:
Adding the dependency seems like the right call here

Reply:
@carllin -- as you might be aware that from Trent we intentionally keep these modules internal. The alternative is use spl-token which will requires us to unpack the whole spl-token account -- which is probably unnecessarily more expensive as I am only interested in the mint and owner field. Can I get a pass to merge this as is to unblock? Thanks!

Reply:
How about working upstream to put this kind of partial-unpacking functionally into spl-token, so you can import that crate from here. This self-inflicted duplication seems quite avoidable. @joncinque's a good resource for guidance on the spl-token crate side